File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):
    answer = []
    
    for i in range(l, r + 1):
        num_str = str(i)
        valid = True
        for j in num_str:
            if j != "0" and j != "5":
                valid = False
                break
        if valid:
            answer.append(i)
    
    if not answer:
        return [-1]
    
    return answer

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.ArrayList;
public class Solution {
    public int[] solution(int l, int r) {
        ArrayList<Integer> result = new ArrayList<>();
        
        for (int i = l; i <= r; i++) {
            String str = Integer.toString(i);
            boolean isValid = true;
            for (char c : str.toCharArray()) {
                if (c != '0' && c != '5') {
                    isValid = false;
                    break;
                }
            }
            if (isValid) {
                result.add(i);
            }
        }
        
        if (result.isEmpty()) {
            return new int[]{-1};
        }
        
        int[] answer = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            answer[i] = result.get(i);
        }
        
        return answer;
    }
}